File: biliPlusAdmin/src/api/banner.ts
import { request } from '@/utils/request'

export interface Banner {
    id?: number
    title: string
    description?: string
    imageUrl: string
    /** 1-视频 2-外链 3-不跳转 */
    linkType: number
    videoId?: number | null
    linkUrl?: string | null
    sortOrder?: number
    /** 0下线 1上线 */
    status?: number
    createTime?: string
    updateTime?: string
}

export const getBannerList = () => {
    return request.get<Banner[]>('/admin/banners/list')
}

export const createBanner = (data: Banner) => {
    return request.post<Banner>('/admin/banners', data)
}

export const updateBanner = (id: number, data: Banner) => {
    return request.put<Banner>(`/admin/banners/${id}`, data)
}

export const deleteBanner = (id: number) => {
    return request.delete<string>(`/admin/banners/${id}`)
}

File: bankend/BiliPlus/src/main/java/com/biliplus/pojo/entity/Banner.java
package com.biliplus.pojo.entity;

import lombok.Data;

import java.time.LocalDateTime;

/**
 * 首页轮播图表实体类
 * 对应数据库表：banner
 */
@Data
public class Banner {

    private Long id;

    /** 标题 */
    private String title;

    /** 描述 */
    private String description;

    /** 图片URL */
    private String imageUrl;

    /** 跳转类型：1-视频 2-外链 3-不跳转 */
    private Integer linkType;

    /** 关联视频ID（linkType=1） */
    private Long videoId;

    /** 外链（linkType=2） */
    private String linkUrl;

    /** 排序，越小越靠前 */
    private Integer sortOrder;

    /** 状态：0-下线 1-上线 */
    private Integer status;

    private LocalDateTime createTime;

    private LocalDateTime updateTime;
}
